tests/components/shelly/test_binary_sensor.py

```diff
@@ -39,12 +39,16 @@ async def test_block_binary_sensor(
     entity_id = f"{BINARY_SENSOR_DOMAIN}.test_name_channel_1_overpowering"
     await init_integration(hass, 1)
 
-    assert hass.states.get(entity_id).state == STATE_OFF
+    entity = hass.states.get(entity_id)
```

Comment:
The `states` object is a state machine not an entity object, and the `get` method returns a state object not entity:
https://github.com/home-assistant/core/blob/2f1ff5ab95b060d678c1ec0f027460f27c332a02/homeassistant/core.py#L2150

```suggestion
    state = hass.states.get(entity_id)
```

Reply:
We have a mix of entity and state all over, so I tried to align to what make more sense to me from a pure semantic approach.
As we then have "`state.state`" in the next line and it didn't make much sense to me.

Reply:
Did a new search and I found <https://www.home-assistant.io/docs/configuration/state_object/> which reference to `state.state` so this is only me non likeing it.

Will update this PR.